packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -21,6 +22,361 @@ import 'gesture_detector.dart';
 /// [WidgetInspector.selectButtonBuilder].
 typedef Widget InspectorSelectButtonBuilder(BuildContext context, VoidCallback onPressed);
 
+/// A class describing a step along a path through a tree of [DiagnosticsNode]
+/// objects.
+///
+/// This class is used to bundle all data required to display the tree with just
+/// the nodes along a path expanded into a single JSON payload.
+class DiagnosticsPathNode {
+  /// Node at the point in the path this [DiagnosticsPathNode] is describing.
+  final DiagnosticsNode node;
+
+  /// Children of the [node] being described.
+  ///
+  /// This value is cached instead of relying on `node.getChildren()` as that
+  /// method call might create new [DiagnosticsNode] objects for each child
+  /// and we would prefer to use the identical [DiagnosticsNode] for each time
+  /// a node exists in the path.
+  final List<DiagnosticsNode> children;
+
+  /// Index of the child that the path continues on.
+  ///
+  /// Equal to `null` if the path does not continue.
+  final int childIndex;
+
+  /// Creates a full description of a step in a path through a tree of
+  /// [DiagnosticsNode] objects.
+  ///
+  /// The [child] argument must not be null.
+  DiagnosticsPathNode({ this.node, this.children, this.childIndex }) : assert(children != null);
+}
+
+List<DiagnosticsPathNode> _followDiagnosticableChain(List<Diagnosticable> chain, {
+  String name,
+  DiagnosticsTreeStyle style,
+}) {
+  final List<DiagnosticsPathNode> path = <DiagnosticsPathNode>[];
+  if (chain.isEmpty)
+    return path;
+  DiagnosticsNode diagnostic = chain.first.toDiagnosticsNode(name: name, style: style);
+  for (int i = 1; i < chain.length; i += 1) {
+    final Diagnosticable target = chain[i];
+    bool foundMatch = false;
+    final List<DiagnosticsNode> children = diagnostic.getChildren();
+    for (int j = 0; j < children.length; j += 1) {
+      final DiagnosticsNode child = children[j];
+      if (child.value == target) {
+        foundMatch = true;
+        path.add(new DiagnosticsPathNode(
+          node: diagnostic,
+          children: children,
+          childIndex: j,
+        ));
+        diagnostic = child;
+        break;
+      }
+    }
+    assert(foundMatch);
+  }
+  path.add(new DiagnosticsPathNode(node: diagnostic, children: diagnostic.getChildren()));
+  return path;
+}
+
+typedef void InspectorSelectionChangedCallback();
```

Comment:
please make sure all public members are documented. Run `flutter analyze --flutter-repo --dartdocs` to get the list of undocumented public members.

Reply:
Surprisingly that was the only public I missed.